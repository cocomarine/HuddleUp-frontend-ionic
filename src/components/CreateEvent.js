import React, { useState } from "react";
import axios from "axios";
import { useNavigate } from "react-router-dom";
import ScrollToTop from "react-scroll-to-top";
import { IoIosArrowUp } from "react-icons/io";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {  faAngleLeft } from "@fortawesome/free-solid-svg-icons";

import { useAuthContext } from "../hooks/useAuthContext";
import { useEventContext } from "../hooks/useEventContext";
import Alert from "./Alert";

import "../styles/common/titles.css";
import "../styles/common/buttons.css";
import "../styles/create-event.css";

const CreateEvent = () => {
  const { user } = useAuthContext();
  const { dispatch } = useEventContext();

  const navigate = useNavigate();

  const initialState = {
    fields: {
      title: "",
      description: "",
      participants: "",
      total_votes: 0,
      date: "",
      category: "",
      AdminId:"",
    },
    alert: {
      message: "",
      success: false,
    },
  };

  const [fields, setFields] = useState(initialState.fields);
  const [alert, setAlert] = useState(initialState.alert);

  const changeLocation = (redirect) => {
    navigate(redirect, { replace: true });
    // window.location.reload();
  };

  const handleAddEvent = (event) => {
    event.preventDefault();

    axios
    .post("http://localhost:4000/events", fields)
    .then((res) => {
      const eventId = res.data.id;
      
      axios
        .post("http://localhost:4000/userevents", {
          voted__suggestionId: null,
          UserId: fields.AdminId,
          EventId: eventId,
        })
        .then((res) => {
          setAlert({
            message: "Event successfully created.",
            success: true,
          });
        });
      
      dispatch({type: "EVENT_CREATED", payload: res.data })
      changeLocation("/invitefriends");
    })
    .catch((err) => {
      setAlert({
        message: "Server error. Please try again later.",
        success: false,
      });
    });

    setAlert({ message: "", success: false });
  };
  
  const handleFieldChange = (e) => {
    fields.AdminId = user.id;
    setFields({ ...fields, [e.target.name]: e.target.value });
  };

  return (
    <div className="eventPgcontainer">
      <h3 className="eventPgtitle page-title">Create Event</h3>
      <Alert message={alert.message} success={alert.success} />
      <form onSubmit={handleAddEvent}>
        <div className="eventForm">
          <label htmlFor="title" className="eventtitle-label subtitle"> 
            Event Title
          </label>
          <div>
            <input
              type="text"
              name="title"
              placeholder="Enter your event title"
              id="title"
              value={fields.title}
              onChange={handleFieldChange}
              required
            ></input>
          </div>
        </div>
        <div className="description">
          <label htmlFor="description" className="description-label subtitle"> 
          Event Description
          </label>
          <div>
            <textarea 
              name="description" 
              id="description" 
              cols="43" 
              rows="10"
              placeholder="Enter the details of the event"
              value={fields.description}
              onChange={handleFieldChange}
              required
            />
          </div>
        </div>

        <div className="participants">
          <label htmlFor="participants" className="participants-label subtitle"> 
            Participants
          </label>
          <div className="participants-container">
            <input
              type="text"
              name="participants"
              placeholder="Enter names"
              value={fields.participants}
              onChange={handleFieldChange}
            />
          </div>
        </div>

        <div className="catetory">
          <label className="category-label subtitle" id="category" htmlFor="categories">
            Event Category
          </label>
          <div>
            <select className="category-options" onChange={handleFieldChange} name="category">
              <option value="restaurant"> Restaurant </option>
              <option value="coffee-tea">Coffee / Tea </option>
              <option value="drinks">Drinks </option>
              <option value="outdoor">Outdoor </option>
              <option value="cinema-show">Cinema / Show </option>
              <option value="playdate">Playdate</option>
              <option value="other">Other</option>
            </select>
          </div>
        </div>
        <div className="date">
          <label htmlFor="date" className="date-label subtitle"> 
            Event Date
          </label>
          <div>
            <input
              value={fields.date}
              name="date"
              type="date"
              onChange={handleFieldChange}
              required
            ></input>
          </div>
        </div>

        <div className="createEvent">
          <button className="createEvent-Btn link-button" type="submit">Create Event</button>
        </div>
      </form>
      <button
        className="backto-myEvents-Btn link-button"
        onClick={() => {
          changeLocation("/myevents");
        }}
      >
        <FontAwesomeIcon
          icon={faAngleLeft}
          className="back-icon"
          data-testid="back-icon"
        />
        &nbsp; Back to My Events
      </button>
      <ScrollToTop smooth component={<IoIosArrowUp />} />
    </div>
  );
};

export default CreateEvent;