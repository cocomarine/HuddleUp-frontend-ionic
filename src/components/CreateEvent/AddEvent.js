// import React, { useState } from "react";
import axios from "axios";

const AddEvent = (fields, setAlert) => {
  console.log("fields", fields);

  axios
    .post("http://localhost:4000/events", fields)
    .then((res) => {
      const eventId = res.data.id;
      
      axios
        .post("http://localhost:4000/userevents", {
          voted__suggestionId: null,
          UserId: fields.AdminId,
          EventId: eventId,
        })
        .then((res) => {
          setAlert({
            message: "Event successfully create.",
            success: true,
          });
      });
      return res.data;
    })
    .catch((err) => {
      setAlert({
        message: "Server error. Please try again later.",
        success: false,
      });
    });
};

export default AddEvent;
