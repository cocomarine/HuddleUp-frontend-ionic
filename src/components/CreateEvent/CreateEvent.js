import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import ScrollToTop from "react-scroll-to-top";
import { IoIosArrowUp } from "react-icons/io";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {  faAngleLeft } from "@fortawesome/free-solid-svg-icons";

import { useAuthContext } from "../../hooks/useAuthContext";
import Alert from "../Alert";
import AddEvent from "./AddEvent";

import "../../styles/common/titles.css";
import "../../styles/common/buttons.css";
import "../../styles/create-event.css";

const CreateEvent = () => {
  const { user } = useAuthContext();

  const navigate = useNavigate();

  const initialState = {
    fields: {
      title: "",
      description: "",
      total_votes: 0,
      date: "",
      category: "",
      AdminId:"",
    },
    alert: {
      message: "",
      success: false,
    },
  };
  const [alert, setAlert] = useState(initialState.alert);

  const changeLocation = (redirect) => {
    navigate(redirect, { replace: true });
    // window.location.reload();
  };

  const handleAddEvent = (event) => {
    event.preventDefault();
    AddEvent(initialState.fields, setAlert);
    changeLocation("/invitefriends");
    console.log(alert);
    setAlert({ message: "", success: false });

  };
  const handleFieldChange = (e) => {
    e.preventDefault();
    if (initialState.fields.hasOwnProperty(e.target.id)) {
      initialState.fields[e.target.id] = e.target.value;
    }
    if (e.target.name === "category") {
      initialState.fields[e.target.name] = e.target.value;
    }
    initialState.fields.AdminId = user.id;
    console.log(initialState.fields);
  };

  return (
    <div className="eventPgcontainer">
      <h3 className="eventPgtitle page-title">Create Event</h3>
      <Alert message={alert.message} success={alert.success} />
      <form onSubmit={handleAddEvent}>
        <div className="eventForm">
          <label htmlFor="title" className="eventtitle-label subtitle"> 
            Event Title
          </label>
          <div>
            <input
              type="text"
              placeholder="Enter your event title"
              id="title"
              onChange={handleFieldChange}
              required
            ></input>
          </div>
        </div>
        <div className="description">
          <label htmlFor="description" className="description-label subtitle"> 
          Event Description
          </label>
          <div>
            <textarea 
              name="description" 
              id="description" 
              cols="43" 
              rows="10"
              placeholder="Enter the details of the event"
              onChange={handleFieldChange}
              required
            />
          </div>
        </div>
        <div className="date">
          <label htmlFor="date" className="date-label subtitle"> 
            Event Date
          </label>
          <div>
            <input
              type="date"
              id="date"
              onChange={handleFieldChange}
              required
            ></input>
          </div>
        </div>
        <div className="catetory">
          <label className="category-label subtitle" id="category" htmlFor="categories">
            Event Category
          </label>
          <div>
            <select className="category-options" onChange={handleFieldChange} name="category">
              <option value="restaurant"> Restaurant </option>
              <option value="coffee-tea">Coffee / Tea </option>
              <option value="drinks">Drinks </option>
              <option value="outdoor">Outdoor </option>
              <option value="cinema-show">Cinema / Show </option>
              <option value="playdate">Playdate</option>
              <option value="other">Other</option>
            </select>
          </div>
        </div>

        <div className="createEvent">
          <button className="createEvent-Btn link-button" type="submit">Create Event</button>
        </div>
      </form>
      <button
        className="backto-myEvents-Btn link-button"
        onClick={() => {
          changeLocation("/myevents");
        }}
      >
        <FontAwesomeIcon
          icon={faAngleLeft}
          className="back-icon"
          data-testid="back-icon"
        />
        &nbsp; Back to My Events
      </button>
      <ScrollToTop smooth component={<IoIosArrowUp />} />
    </div>
  );
};

export default CreateEvent;