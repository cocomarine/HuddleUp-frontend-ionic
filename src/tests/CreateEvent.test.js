import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import { AuthContextProvider } from "../contexts/AuthContext";
import CreateEvent from "../components/CreateEvent/CreateEvent";

describe("CreateEvent", () => {
  it("renders correctly", () => {
    const rendered = renderer.create(
      <AuthContextProvider>
        <MemoryRouter>
          <CreateEvent />
        </MemoryRouter>
      </AuthContextProvider>
    );
    expect(rendered).toMatchSnapshot();
  });
});