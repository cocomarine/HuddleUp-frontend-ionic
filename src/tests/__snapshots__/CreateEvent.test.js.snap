// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateEvent renders correctly 1`] = `
<div
  className="eventPgcontainer"
>
  <h1
    className="eventPgtitle"
  >
    Create Event
  </h1>
  <div
    className="error"
  />
  <form
    onSubmit={[Function]}
  >
    <div
      className="eventForm"
    >
      <label
        htmlFor="title"
      >
         Event Title:
      </label>
      <div>
        <input
          id="title"
          onChange={[Function]}
          placeholder="Enter your event title"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="description"
    >
      <label
        htmlFor="description"
      >
         Event Description:
      </label>
      <div>
        <input
          id="description"
          onChange={[Function]}
          placeholder="Enter the details of the event"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="date"
    >
      <label
        htmlFor="date"
      >
         Event Date:
      </label>
      <div>
        <input
          id="date"
          onChange={[Function]}
          required={true}
          type="date"
        />
      </div>
    </div>
    <br />
    <label
      className="category"
      htmlFor="categories"
      id="category"
    >
      Event Category
    </label>
    <select
      name="category"
      onChange={[Function]}
    >
      <option
        value="restaurant"
      >
         Restaurant 
      </option>
      <option
        value="coffee-tea"
      >
        Coffee / Tea 
      </option>
      <option
        value="drinks"
      >
        Drinks 
      </option>
      <option
        value="outdoor"
      >
        Outdoor 
      </option>
      <option
        value="cinema-show"
      >
        Cinema / Show 
      </option>
      <option
        value="playdate"
      >
        Playdate
      </option>
      <option
        value="other"
      >
        Other
      </option>
    </select>
    <div
      className="createEvent"
    >
      <button
        type="submit"
      >
        Create Event
      </button>
    </div>
  </form>
  <button
    className="backto-myEventsBtn"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-angle-left back-icon"
      data-icon="angle-left"
      data-prefix="fas"
      data-testid="back-icon"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 256 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256 214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Â  Back to My Events
  </button>
</div>
`;
